tests/virtctl/ssh.go

```diff
@@ -0,0 +1,94 @@
+package virtctl
+
+import (
+	"crypto/ecdsa"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"golang.org/x/crypto/ssh"
+
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/clientcmd"
+	"kubevirt.io/kubevirt/tests/console"
+	"kubevirt.io/kubevirt/tests/flags"
+	"kubevirt.io/kubevirt/tests/libvmi"
+	"kubevirt.io/kubevirt/tests/util"
+)
+
+var _ = Describe("[sig-compute][virtctl]SSH", func() {
+	var pub ssh.PublicKey
+	var keyFile string
+	var virtClient kubecli.KubevirtClient
+
+	BeforeEach(func() {
+		var err error
+		virtClient, err = kubecli.GetKubevirtClient()
+		Expect(err).ToNot(HaveOccurred())
+		// Disable SSH_AGENT to not influence test results
+		Expect(os.Setenv("SSH_AUTH_SOCK", "/dev/null")).To(Succeed())
+		keyFile = filepath.Join(GinkgoT().TempDir(), "id_rsa")
+		tests.BeforeTestCleanup()
+		var priv *ecdsa.PrivateKey
+		priv, pub, err = NewKeyPair()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(DumpPrivateKey(priv, keyFile)).To(Succeed())
+	})
+
+	It("should succeed to execute a command on the VM using the ssh native method", func() {
+		By("injecting a SSH public key into a VMI")
+		vmi := libvmi.NewCirros(
+			libvmi.WithCloudInitNoCloudUserData(renderUserDataWithKey(pub), false))
+		vmi, err := virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(vmi)
+		Expect(err).ToNot(HaveOccurred())
+
+		vmi = tests.WaitUntilVMIReady(vmi, console.LoginToCirros)
+		By("ssh into the VM using the native option")
+		Expect(clientcmd.NewRepeatableVirtctlCommand(
+			"ssh",
+			"--local-ssh=false",
+			"--namespace", util.NamespaceTestDefault,
+			"--username", "cirros",
+			"--identity-file", keyFile,
+			"--known-hosts=",
```

Comment:
Just wondering: is this arg empty on purpose? maybe then better to drop it?

Reply:
yes, it is on purpose. It avoids the host key check like for the [scp command](https://github.com/kubevirt/kubevirt/blob/main/tests/virtctl/scp.go#L67)

Reply:
We could consider a `--strict-host-key-checking` flag like `gcloud` has. That would probably be more user friendly :)